drivers/device_uid/device_uid_handlers.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2018 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <syscall_handler.h>
+#include <device_uid.h>
+
+Z_SYSCALL_HANDLER(device_uid_get, dev, buffer, length) {
+
+	Z_OOPS(Z_SYSCALL_DRIVER_DEVICE_UID(dev, device_uid_get));
+	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(buffer, length));
+
+	return _impl_device_uid_get((struct device *)dev, buffer, length);
+}
+
+Z_SYSCALL_HANDLER(device_uid_get_length, dev) {
```

Comment:
Why do we need 2 calls instead of one?

Reply:
To get the length of the UID (differs for different vendors/cpus) before reading it. This is maybe useful if you have to decide between tow sources of ids (take the longer one) or so.